File: src/algorithms/sort-array-by-parity.js
/**
 * Given an integer array nums, returns the array after moving all the even
 * integers to the beginning of the array and all the odd integers to the back.
 * 
 * @param {number[]} nums - An array of numbers
 * @returns {number[]} - Returns the modified array
 */
function sortArrayByParity (nums) {
  let i = 0;
  let j = nums.length - 1;
  while (i < j) {
    if (nums[i] % 2 === 1) {
      if (nums[j] % 2 === 0) {
        let temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
      } else {
        j--;
      }
    } else {
      i++;
    }
  }
  return nums;
}

File: src/algorithms/sort-array-by-parity.py
def sortArrayByParity(nums):
    """Given an integer array nums, move all the even integers at the beginning 
    of the array followed by all the odd integers. Return any array that 
    satisfies this condition.

    Args:
        nums (list[int]): A list of numbers.

    Returns:
        list[int]: Returns a sorted list of numbers.
    """
    res = []
    stack = []
    for num in nums:
        if num % 2 == 0:
            res.append(num)
        else:
            stack.append(num)
    for num in stack:
        res.append(num)
    return res
